core/src/main/java/hudson/security/SidACL.java

```diff
@@ -66,8 +73,15 @@ public boolean hasPermission(@Nonnull Authentication a, Permission permission) {
      *      Otherwise null, indicating that this ACL doesn't have any entry for it.
      */
     protected Boolean _hasPermission(@Nonnull Authentication a, Permission permission) {
+        PrincipalSid principalSid;
+        try {
+            principalSid =  principalSidCache.get(a, () -> {return new PrincipalSid(a);});
+        } catch (ExecutionException e) {
+            LOGGER.log(WARNING, "Failed to populate Principal Sid Cache", e);
+            principalSid = new PrincipalSid(a);
```

Comment:
It is quite unlikely that it passes after the exception above

Reply:
Agreed, I'm not certain what the right behaviour is in this case. Probably just returning is best.